eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalStatements.scala

```diff
@@ -46,7 +48,7 @@ object JournalStatements {
 
 
   trait InsertRecords[F[_]] {
-    def apply(key: Key, segment: SegmentNr, events: Nel[EventRecord]): F[Unit]
+    def apply[A](key: Key, segment: SegmentNr, events: Nel[EventRecord[A]])(implicit W: EventualWrite[F, A]): F[Unit]
```

Comment:
you think this is the layer we should do reads/writes?

Reply:
Hmmm, now that I think about it, perhaps we could pass already written payload here (smth like EventRecord[EventualPayloadAndType]), and have the actual writing be done on ReplicateRecords level, and reading - somewhere on Journal level (same as KafkaRead is done now). Thanks for pointing it out!